python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/asset_decorator.py

```diff
@@ -0,0 +1,135 @@
+from typing import Any, Callable, Iterable, List, Mapping, Optional
+
+from dagster import (
+    AssetsDefinition,
+    AssetSpec,
+    BackfillPolicy,
+    FreshnessPolicy,
+    PartitionsDefinition,
+    multi_asset,
+)
+from dagster._core.definitions.auto_materialize_policy import AutoMaterializePolicy
+
+from dagster_embedded_elt.sling.dagster_sling_translator import DagsterSlingTranslator
+from dagster_embedded_elt.sling.sling_replication import SlingReplicationParam, validate_replication
+
+
+def get_streams_from_replication(replication_config: Mapping[str, Any]) -> Iterable[str]:
+    """Returns a list of streams from a Sling replication config.
+
+    If an `asset_key` is provided, it will be used instead. For example, below the stream name
+    will be read as `public.foo_users`
+
+
+    .. code-block:: yaml
+
+        streams:
+          public.users:
+            meta:
+              dagster:
+                asset_key: public.foo_users
+
+    Args:
+        replication_config: Mapping[str, Any]: A dictionary of a Sling replication config.
+    """
+    keys = []
+
+    for key, value in replication_config["streams"].items():
+        if isinstance(value, dict):
+            asset_key_config = value.get("meta", {}).get("dagster", {}).get("asset_key", [])
+            keys.append(asset_key_config if asset_key_config else key)
+        else:
+            keys.append(key)
+    return keys
+
+
+def sling_assets(
+    *,
+    replication_config: SlingReplicationParam,
+    dagster_sling_translator: DagsterSlingTranslator = DagsterSlingTranslator(),
+    partitions_def: Optional[PartitionsDefinition] = None,
+    backfill_policy: Optional[BackfillPolicy] = None,
+    op_tags: Optional[Mapping[str, Any]] = None,
+    group_name: Optional[str] = None,
+    code_version: Optional[str] = None,
+    freshness_policy: Optional[FreshnessPolicy] = None,
+    auto_materialize_policy: Optional[AutoMaterializePolicy] = None,
+) -> Callable[..., AssetsDefinition]:
+    """Create a definition for how to materialize a Sling replication stream as a Dagster Asset, as
+    described by a Sling replication config. This will create on Asset for every Sling target stream.
+
+    A Sling Replication config is a configuration that maps sources to destinations. For the full
+    spec and descriptions, see `Sling's Documentation <https://docs.slingdata.io/sling-cli/run/configuration>`_.
+
+    Args:
+        replication_config: SlingReplicationParam: A path to a Sling replication config, or a dictionary of a replication config.
```

Comment:
I would use the fully expanded type here, since this type alias is internally facing.